src/pid/maps.rs

```diff
@@ -0,0 +1,243 @@
+//! Process memory mappings information from `/proc/[pid]/maps`.
+
+use std::ffi::OsString;
+use std::io::{self, BufRead};
+use std::os::unix::ffi::OsStringExt;
+use std::path::PathBuf;
+use std::{fs, ops};
+
+use libc;
+use nom::{rest, space};
+
+use parsers::{map_result, parse_usize_hex, parse_u32_hex, parse_u64, parse_u64_hex};
+use unmangle::unmangled_path;
+
+/// Process memory mapping information.
+///
+/// Due to the way paths are encoded by the kernel before exposing them in
+/// `/proc/[pid]/maps`, the parsing of `path` and `is_deleted` is
+/// ambiguous. For example, all the following path/deleted combinations:
+///
+/// - `/tmp/a\nfile` *(deleted file)*
+/// - `/tmp/a\nfile (deleted)` *(existing file)*
+/// - `/tmp/a\\012file` *(deleted file)*
+/// - `/tmp/a\\012file (deleted)` *(existing file)*
+///
+/// will be mangled by the kernel and decoded by this module as:
+///
+/// ```rust,ignore
+/// MemoryMapping (
+///    ...,
+///    path: PathBuf::from("/tmp/a\nfile"),
+///    is_deleted:true,
+/// )
+/// ```
+///
+/// If the `path` of a mapping is required for other than purely informational
+/// uses (such as opening and/or memory mapping it), a more reliable source
+/// (such as `/proc/[pid]/map_files`) should be used, if available. The open
+/// file `(dev, inode)` should also be checked against the values provided by
+/// the mapping.
+///
+/// See `man 5 proc`, `Linux/fs/proc/task_mmu.c`, and `Linux/fs/seq_file.c`.
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub struct MemoryMapping {
```

Comment:
How do you feel about `MemoryMap` as opposed to `MemoryMapping`?  It's a bit shorter, and matches `maps` more closely.